### PR TITLE
Feature/lappeenranta

### PR DESCRIPTION
**Pull request by jtuomist**
Add baseline_year to datasets. Is this OK?
lappeenranta-nzc is an independent instance so it should not interfere with other instances.

### CODE REVIEW

**Comment by jtuomist** [comment]
@mechenich not urgent, but can you take a look at the unit problem that occurs with units without numerator, e.g. 1/a.

File: configs/frameworks/health.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ nodes:
 
 # -------------- Emissions --------------
 
-- id: transport_emission_factor
+- id: transport_emission_factor2
```
[the remark this answers is not in the document]
Comment (jtuomist) [reply configs/frameworks/health.yaml:17]: This relates to the fact that health.yaml is a file that is used by several KP instances. Therefore, the names must match but not overlap with the names in several other files. Suggestions on how to do this systematically and unambiguously are welcome.

File: nodes/datasets.py [hunk 2]
```diff
@@ -199,8 +202,22 @@ def load(self, context: Context) -> ppl.PathsDataFrame:
...
+            if YEAR_COLUMN not in df.primary_keys:
+                df = df.add_to_index(YEAR_COLUMN)
+            if len(df.filter(pl.col(YEAR_COLUMN).lt(100))) > 0:
+                baseline_year = context.get_parameter_value('baseline_year', required=True)
+                df = df.with_columns(
+                    pl.when(pl.col(YEAR_COLUMN).lt(100))
+                    .then(pl.col(YEAR_COLUMN) + baseline_year)
+                    .otherwise(pl.col(YEAR_COLUMN)).alias(YEAR_COLUMN),
+                )
+                df = df.with_columns(pl.col(YEAR_COLUMN).cast(int).alias(YEAR_COLUMN))
```
Comment (bbliem) [inline nodes/datasets.py:215]: I believe the cast be merged into the previous statement.

File: nodes/datasets.py [hunk 3]
```diff
@@ -199,8 +202,22 @@ def load(self, context: Context) -> ppl.PathsDataFrame:
...
 
-        if YEAR_COLUMN in cols and YEAR_COLUMN not in df.primary_keys:
-            df = df.add_to_index(YEAR_COLUMN)
+        if YEAR_COLUMN in cols:
+            if YEAR_COLUMN not in df.primary_keys:
+                df = df.add_to_index(YEAR_COLUMN)
+            if len(df.filter(pl.col(YEAR_COLUMN).lt(100))) > 0:
+                baseline_year = context.get_parameter_value('baseline_year', required=True)
+                df = df.with_columns(
+                    pl.when(pl.col(YEAR_COLUMN).lt(100))
```
Comment (bbliem) [inline nodes/datasets.py:211]: I only have limited Polars knowledge, but it seems that this is equivalent and maybe easier to read: `pl.when(pl.col(YEAR_COLUMN) < 100)`

File: nodes/datasets.py [hunk 4]
```diff
@@ -199,8 +202,22 @@ def load(self, context: Context) -> ppl.PathsDataFrame:
...
+                    .then(pl.col(YEAR_COLUMN) + baseline_year)
+                    .otherwise(pl.col(YEAR_COLUMN)).alias(YEAR_COLUMN),
+                )
+                df = df.with_columns(pl.col(YEAR_COLUMN).cast(int).alias(YEAR_COLUMN))
+
+                # Duplicates may occur when baseline year overlaps with existing data points.
+                meta = df.get_meta()
+                df = ppl.to_ppdf(df.unique(subset = meta.primary_keys, keep = 'last',
+                                           maintain_order = True), meta = meta)
```
Comment (bbliem) [inline nodes/datasets.py:220]: By convention, there should be no spaces around `=` when setting named parameters.

**Review by bbliem — COMMENTED** [review]
I am only vaguely familiar with KP at this point, but here are some things I could find.

**Pull request merged**